WordPress/Classes/ViewRelated/Site Creation/Web Address/DesignSystem/UIColor+DesignSystem.swift

```diff
@@ -0,0 +1,44 @@
+import UIKit
+
+/// Replica of the `Color` structure
```

Comment:
Indeed. I couldn't think of a way since they are separate API's. One option would be to leave it in `Color` and use it from UIKit by initiating `UIColor` from `SwiftUI.Color`. But the documentation advises to not overuse those initializers and I have found them to not work perfectly in all situations.

It would be nice if we can find a way to not duplicate.

Reply:
Okay, I see.

Yes, in this case we would need to have something additional to keep using the main initializers of `UIColor` and `Color` but to ensure consistency. 

Maybe some sort of wrapper value, to have a single definition of all the colors?

```
struct DSColor {
   let name: String
   
   var color: Color {
       return Color(name)
   }
   
   var uiColor: UIColor {
       return UIColor(named: name)
   }
}
```

Or just a `string` extension to have the names strongly typed? My concern is the duplication of all these color names since we want to be as clear as possible with the definition of the Design System.

Reply:
> Or just a string extension to have the names strongly typed
^ This change is now implemented as per your suggestion ^


With the suggestion, we wouldn't be able to benefit from IDE's infering mechanism. Say you're in a SwiftUI View and you want to assing a color:

```Swift
.foregroundColor(DSColor.Foreground.primary)
```
and in a UIKit view
```Swift
label.textColor = DSColor.Foreground.uiPrimary
```

So in both cases, you would have a list of duplicate values that return different types.

I think your concern about duplication is totally valid. Though I don't think it would take away from being clear with the Design System. It increases the likelihood that we might have a mismatch between the same color from 2 api's.
They get their values from the same color set though. So the extension layer is duplicated but their values are not. The only error prone part would be when we're adding the extensions and use the wrong color asset.

This is the best version I could find but it is by no means a great solution. The same will apply to Fonts too for instance. 

My logic is, since we'll define these values once and not keep adding on them frequently (as the design system will be defined), the possibility of introducing a mismatch would be only on the initial stage. It can be caught by testing and reviews. The suggestion you've posted above still has the same possibility. So I would favor having a clear usage over that one.